File: python/1.two-sum.py
class Solution(object):
    def solve(self, nums, target):
        marked = {} # key:num
        for i, num1 in enumerate(nums):
            num2 = target - num1
            if num2 in marked:
                return [i, marked[num2]]
            marked[num1]=i


    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        return self.solve(nums, target)

File: cpp/1.two-sum.cpp
class Solution {
public:
    vector<int> solve(vector<int> &nums, int target){
	   vector<int> res;
	   int n = nums.size();
	   for(int i = 0 ;i < n;i++){
		  for(int j = i+1; j < n; j++){
			  if(nums[i]+nums[j]==target){
				  return {i,j};
			  }
		  }
	   }
	   return res;
    }
    vector<int> twoSum(vector<int>& nums, int target) {
	   return solve(nums, target);
    }
};
